Added method wrap()

islas27 opened the pull request:
- Added unit test
- Added documentation
- Added examples
- Improved documentation about cat & fixed examples

Closes #5

islas27 committed:
Added method wrap()

— added unit test
— WIP: breaks a test in cat (helper available inside a function passed as a parameter)
- Issue #5

islas27 committed:
Fixed broken test

- WIP (#5): Refactoring & adding tests

ryanez commented on index.js:
```diff
@@ -6,19 +6,33 @@ const isType = (type, inp) => typeof inp === type
-      if (isFunc(arg)) this.cat(arg.apply(this))
-      else if (isArr(arg)) this.cat(...arg.flatten())
+      let arg = args[i]
+      if (isFunc(arg)) ccat(arg.apply(that))
```
the `apply` shall be used when you don't know the amount of arguments and want to send an array, in this case you know there are NO arguments, use `call` instead.

islas27 committed:
Added documentation for wrap()

- Returned reported from spec to nyan
- Added info to README
- Added examples on how to use wrap()
- Fixed examples about cat()
- Some improvements in index.js readability
- Improved readibility in a test in wrap.spec
- Added test in cat.spec
